### PR TITLE
Adding Support for 4 new cloudflare_teams_* resources and additional README.md Improvements

### PR DESCRIPTION
**Pull request by devinbfergy**
Adds support for the following

- cloudflare_teams_list
- cloudflare_teams_location
- cloudflare_teams_proxy_endpoint
- cloudflare_teams_rule

Also the PR adds some information about a suggested way to perform local testing.

Adding the cloudflare_teams_account will need to be done in future MR.

### CODE REVIEW

File: internal/app/cf-terraforming/cmd/generate.go [hunk 1]
```diff
@@ -879,6 +880,106 @@ func generateResources() func(cmd *cobra.Command, args []string) {
 				if err != nil {
 					log.Fatal(err)
 				}
+			case "cloudflare_teams_list":
+				jsonPayload, _, err := api.ListTeamsLists(context.Background(), &cloudflare.ResourceContainer{Identifier: accountID}, cloudflare.ListTeamListsParams{})
+				if err != nil {
+					log.Fatal(err)
+				}
+				// creating our own listItems struct because Items nees to be list of string to match
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:888]: check out the prior art on remapping responses - https://github.com/cloudflare/cf-terraforming/pull/669/files#diff-c8a271423d325a18f64e8eb426bae4c393315dfd1fbc9bad21916c9eead1647aL852

File: README.md [hunk 2]
```diff
@@ -275,6 +279,41 @@ test. The Terraform files then allow us to build what the resource structure is
 expected to look like and once the tool parses the API response, we can compare
 that to the static file.
 
+Suggested local testing steps:
+
+1. Create a main file (do not commit this up)
```
Comment (jacobbednarz) [inline README.md:284]:
```suggestion
1. Create a file with the basic provider configuration (do not commit this file)
```

File: README.md [hunk 3]
```diff
@@ -275,6 +279,41 @@ test. The Terraform files then allow us to build what the resource structure is
...
+
+      ```bash
```
Comment (jacobbednarz) [inline README.md:286]: can you outdent these please? we don't need indentation + code blocks in markdown so let's keep it consistent with the existing format of outdented + code fences.

File: README.md [hunk 4]
```diff
@@ -275,6 +279,41 @@ test. The Terraform files then allow us to build what the resource structure is
...
+                  EOF
+      ```
+
+2. Initialize terraform
+
+      ```bash
+      terraform init
+      ```  
+
+3. Run tests (Cloudflare Install path should be path to repository)
```
Comment (jacobbednarz) [inline README.md:305]: if you initiliase `terraform` in the same directory as `cf-terraforming`, this shouldn't be needed as it defaults to `.`.

File: README.md [hunk 5]
```diff
@@ -275,6 +279,41 @@ test. The Terraform files then allow us to build what the resource structure is
...
+
+    ```bash
+    CLOUDFLARE_TERRAFORM_INSTALL_PATH=~/gh/cf-terraforming make test
+    ```
+
+If you want to run a specific test case you can do so with the TESTARGS variable and -run flag
+
+```bash
+CLOUDFLARE_TERRAFORM_INSTALL_PATH=~/gh/cf-terraforming TESTARGS="-run '^TestResourceGeneration/cloudflare_teams_list'" make test
```
Comment (jacobbednarz) [inline README.md:314]: similar to https://github.com/cloudflare/cf-terraforming/pull/669/files#r1451844016

File: internal/app/cf-terraforming/cmd/generate.go [hunk 6]
```diff
@@ -879,6 +880,106 @@ func generateResources() func(cmd *cobra.Command, args []string) {
 				if err != nil {
 					log.Fatal(err)
 				}
+			case "cloudflare_teams_list":
+				jsonPayload, _, err := api.ListTeamsLists(context.Background(), &cloudflare.ResourceContainer{Identifier: accountID}, cloudflare.ListTeamListsParams{})
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:884]: we already conditionally build the resource identifier for reuse

```suggestion
				jsonPayload, _, err := api.ListTeamsLists(context.Background(), identifier, cloudflare.ListTeamListsParams{})
```

File: internal/app/cf-terraforming/cmd/generate.go [hunk 7]
```diff
@@ -879,6 +880,106 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
+					Items       []string        `json:"items,omitempty"`
+					Count       uint64          `json:"count,omitempty"`
+					CreatedAt   *time.Time      `json:"created_at,omitempty"`
+					UpdatedAt   *time.Time      `json:"updated_at,omitempty"`
+				}
+				// get items and set them to the object in the struct
+				for _, cfList := range jsonPayload {
+					items_struct, _, err := api.ListTeamsListItems(
+						context.Background(), 
+					    cloudflare.AccountIdentifier(accountID),
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:904]:
```suggestion
					    identifier, 
```

File: internal/app/cf-terraforming/cmd/generate.go [hunk 8]
```diff
@@ -879,6 +880,106 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
+				if err != nil {
+					log.Fatal(err)
+				}
+				err = json.Unmarshal(m, &jsonStructData)
+				if err != nil {
+					log.Fatal(err)
+				}
+				resourceCount = len(tfTeamsList)	
+			case "cloudflare_teams_location":
+				jsonPayload, _, err := api.TeamsLocations(context.Background(), accountID)
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:944]:
```suggestion
				jsonPayload, _, err := api.TeamsLocations(context.Background(), identifier)
```

File: internal/app/cf-terraforming/cmd/generate.go [hunk 9]
```diff
@@ -879,6 +880,106 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
+					log.Fatal(err)
+				}
+				resourceCount = len(jsonPayload)
+				m, _ := json.Marshal(jsonPayload)
+				err = json.Unmarshal(m, &jsonStructData)
+				if err != nil {
+					log.Fatal(err)
+				}
+			case "cloudflare_teams_proxy_endpoint":
+				jsonPayload, _, err := api.TeamsProxyEndpoints(context.Background(), accountID)
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:955]:
```suggestion
				jsonPayload, _, err := api.TeamsProxyEndpoints(context.Background(), identifier)
```

File: internal/app/cf-terraforming/cmd/import.go [hunk 10]
```diff
@@ -368,6 +372,50 @@ func runImport() func(cmd *cobra.Command, args []string) {
 					log.Fatal(err)
 				}
 
+				m, _ := json.Marshal(jsonPayload)
+				err = json.Unmarshal(m, &jsonStructData)
+				if err != nil {
+					log.Fatal(err)
+				}
+			case "cloudflare_teams_list":
+				jsonPayload, _, err := api.ListTeamsLists(context.Background(), &cloudflare.ResourceContainer{Identifier: accountID}, cloudflare.ListTeamListsParams{})
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/import.go:381]:
```suggestion
				jsonPayload, _, err := api.ListTeamsLists(context.Background(), identifier, cloudflare.ListTeamListsParams{})
```

File: testdata/cloudflare/cloudflare_teams_list.yaml [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+      Content-Type:
+      - application/json
+    url: https://api.cloudflare.com/client/v4/accounts/f037e56e89293a057740de681ac9abbe/gateway/lists/971fc4e8-388e-4ab9-b377-16430c0fc018/items?page=1&per_page=50
+    method: GET
+  response:
+    body: |
+      {
+        "result": [
+          {
+            "value": "cloudfront.net",
```
Comment (jacobbednarz) [inline testdata/cloudflare/cloudflare_teams_list.yaml:57]: let's swap this to `example.com`

**Review by jacobbednarz — CHANGES_REQUESTED** [review]
solid start! few things to address but overall, looking good. thanks!

File: README.md (hunk 4 above)
Comment (devinbfergy) [reply README.md:305]: I actually tried this an it still failed. Potentially its not set to the root directory of the project. I am going to just add it to the make file and we will see if it causes issues or if you don't like that solution.

File: internal/app/cf-terraforming/cmd/generate.go [hunk 12]
```diff
@@ -879,11 +879,89 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
+				err = json.Unmarshal(m, &jsonStructData)
+				if err != nil {
+					log.Fatal(err)
+				}
+				// check for empty descriptions
+				for i := 0; i < resourceCount; i++ {
+					if jsonStructData[i].(map[string]interface{})["description"] == "" {
+						jsonStructData[i].(map[string]interface{})["description"] = "default"
+					}
+
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:958]:
```suggestion
```

**Comment by jacobbednarz** [comment]
thanks @devinbfergy 👏 looks good from me!

**Pull request merged**